tests/ccgbugs/tseq_construction_bug.nim

```diff
@@ -0,0 +1,20 @@
+discard """
+  description: '''
+    Regression test for a C code generator bug with `seq` constructions that
+    resulted in a heap corruption at run-time.
+    Reduced from https://github.com/nim-works/nimskull/issues/1117"
+  '''
+  targets: "c"
+  matrix: "-d:useMalloc"
+  valgrind: true
+"""
+
+type Sequence[T] = seq[T]
+
+# tcompile-time-evaluated value must have a generic instantiation as the type
```

Comment:
```suggestion
# compile-time-evaluated value must have a generic instantiation as the type
```

typo?

Reply:
Ah, no, I erroneously cut off parts of the "the" when shortening the text.
```suggestion
# the compile-time-evaluated value must have a generic instantiation as the type
```